src/test/java/com/google/starfish/services/FavoriteNoteServiceTest.java

```diff
@@ -0,0 +1,300 @@
+package com.google.starfish.services;
+
+import java.util.Date;
+import java.util.Calendar;
+import java.sql.SQLException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import javax.sql.DataSource;
+import static com.ninja_squad.dbsetup.Operations.*;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import com.google.starfish.services.FavoriteNoteService.Recency;
+import com.google.starfish.models.Note;
+import com.ninja_squad.dbsetup.DbSetup;
+import com.ninja_squad.dbsetup.destination.DataSourceDestination;
+import com.ninja_squad.dbsetup.operation.Operation;
+
+@RunWith(JUnit4.class)
+public class FavoriteNoteServiceTest {
+  private Constants constants = new Constants();
+  private String NOTES = constants.NOTES;
+  private String LABELS = constants.LABELS;
+  private String FAVORITE_NOTES = constants.FAVORITE_NOTES;
+  private String USERS = constants.USERS;
+  private DataSource pool = Constants.pool;
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+
+  private final Date today = favoriteNoteService.getDateBasedOnRecency(Recency.TODAY);
+  private final Date thisWeek = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_WEEK);
+  private final Date thisMonth = favoriteNoteService.getDateBasedOnRecency(Recency.THIS_MONTH);
+  private final Date allTime = favoriteNoteService.getDateBasedOnRecency(Recency.ALL_TIME);
```

Comment:
I'm admittedly not as familiar anymore with JUnit testing, but...  Typically, access modifiers should not be changed just for tests, and it seems like you've changed the access modifier in the class above to be `protected` so that you can use the private method here.  Those access modifiers should be used for the sole purpose of meeting the needs of the API as designed to interact with other classes, etc.  

Regardless, when testing, data like this used in an assertion should actually likely be hardcoded rather than derived from some method.  You can imagine that if the method is faulty, testing some functionality that uses it would return the same faulty result as this expected value, and so, your test for this functionality would pass even if the method is faulty.  Rather, it would be much clearer, if it was made explicitly clear what values you're using to make assertions about expected vs. derived (from whatever functionality you're testing).

Reply:
This makes so much sense and I definitely thought about this when I changed the access modifier of the `getDateBasedOnRecency` method to work in the test class. I think I was trying to make the date generated dynamic, i.e. `today` would always be in reference to the actual date today when the test is run, but in hindsight, there really isn't any need for that and the way I've done it here, if the `getDateBasedOnRecency` method was faulty, I would've never known just from these unit tests because they implicitly assume that it's 100% functional. I'll hardcode the dates!